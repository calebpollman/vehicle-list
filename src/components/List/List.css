.list-container {
  background-color: none;
  height: 86%;
  overflow: scroll;
}