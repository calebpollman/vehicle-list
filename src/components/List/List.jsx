import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '../ListItem/ListItem';
import {LoadIcon} from '../Icons/Icons';
import './List.css';

const List = ({listLoading, searchActive, searchResults, vehicles}) => {
  
  const renderVehicles = (vehicles) => {
    return vehicles.map(vehicle => {
      const vehicleTitle = `${vehicle.year} ${vehicle.make} ${vehicle.model}`;
      return (
        <ListItem 
          key={vehicle.id}
          vehicle={vehicle}
          vehicleTitle={vehicleTitle}
        />
      );
    });
  }
  
  listLoading ? 
  vehicles = <LoadIcon /> : searchActive ? 
  vehicles = renderVehicles(searchResults) : vehicles = renderVehicles(vehicles);
  
  return (
    <ul className="list-container">
      {vehicles}
    </ul>
  );
}

List.propTypes = {
  listLoading: PropTypes.bool.isRequired,
  searchActive: PropTypes.bool.isRequired,
  searchResults: PropTypes.array.isRequired,
  vehicles: PropTypes.array.isRequired,
}

export default List;
