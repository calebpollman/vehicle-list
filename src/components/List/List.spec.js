import React from 'react';
import {shallow} from 'enzyme';
import List from './List';

describe('List', () => {
  let wrapper;
  const vehicles = [
    {
      "id": 0,
      "year" : "2013",
      "make" : "Kia",
      "model" : "Optima",
      "mileage" : 24235,
      "drivetrain" : "FWD",
      "bodytype" : "sedan",
      "image_url" : "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
      "created_at" : "2016-10-14T20:13:22.586Z"
    },
    {
      "id": 1,
      "year" : "2013",
      "make" : "Hyundai",
      "model" : "Accent",
      "mileage" : 21587,
      "drivetrain" : "FWD",
      "bodytype" : "sedan",
      "image_url" : "http://www.conceptcarz.com/images/Hyundai/2013-Hyundai-Accent-Sedan-Image-01.jpg",
      "created_at" : "2016-11-14T20:13:22.586Z"
    },
  ];

  beforeEach(() => {
    wrapper = shallow(
      <List 
        listLoading={false} 
        searchActive={false} 
        searchResults={[]} 
        vehicles={vehicles} 
      />
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a wrapper <ul />', () => {
    expect(wrapper.find('ul').length).toEqual(1);
  });

  it('should render an instance of <ListItem /> for each index of vehicles', () => {
    expect(wrapper.find('ListItem').length).toEqual(vehicles.length);
  });
});