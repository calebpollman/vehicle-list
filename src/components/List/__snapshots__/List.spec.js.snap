// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    listLoading={false}
    searchActive={false}
    searchResults={Array []}
    vehicles={
        Array [
            Object {
              "bodytype": "sedan",
              "created_at": "2016-10-14T20:13:22.586Z",
              "drivetrain": "FWD",
              "id": 0,
              "image_url": "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
              "make": "Kia",
              "mileage": 24235,
              "model": "Optima",
              "year": "2013",
            },
            Object {
              "bodytype": "sedan",
              "created_at": "2016-11-14T20:13:22.586Z",
              "drivetrain": "FWD",
              "id": 1,
              "image_url": "http://www.conceptcarz.com/images/Hyundai/2013-Hyundai-Accent-Sedan-Image-01.jpg",
              "make": "Hyundai",
              "mileage": 21587,
              "model": "Accent",
              "year": "2013",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ListItem
          vehicle={
                    Object {
                              "bodytype": "sedan",
                              "created_at": "2016-10-14T20:13:22.586Z",
                              "drivetrain": "FWD",
                              "id": 0,
                              "image_url": "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
                              "make": "Kia",
                              "mileage": 24235,
                              "model": "Optima",
                              "year": "2013",
                            }
          }
          vehicleTitle="2013 Kia Optima"
/>,
        <ListItem
          vehicle={
                    Object {
                              "bodytype": "sedan",
                              "created_at": "2016-11-14T20:13:22.586Z",
                              "drivetrain": "FWD",
                              "id": 1,
                              "image_url": "http://www.conceptcarz.com/images/Hyundai/2013-Hyundai-Accent-Sedan-Image-01.jpg",
                              "make": "Hyundai",
                              "mileage": 21587,
                              "model": "Accent",
                              "year": "2013",
                            }
          }
          vehicleTitle="2013 Hyundai Accent"
/>,
      ],
      "className": "list-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "vehicle": Object {
            "bodytype": "sedan",
            "created_at": "2016-10-14T20:13:22.586Z",
            "drivetrain": "FWD",
            "id": 0,
            "image_url": "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
            "make": "Kia",
            "mileage": 24235,
            "model": "Optima",
            "year": "2013",
          },
          "vehicleTitle": "2013 Kia Optima",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "vehicle": Object {
            "bodytype": "sedan",
            "created_at": "2016-11-14T20:13:22.586Z",
            "drivetrain": "FWD",
            "id": 1,
            "image_url": "http://www.conceptcarz.com/images/Hyundai/2013-Hyundai-Accent-Sedan-Image-01.jpg",
            "make": "Hyundai",
            "mileage": 21587,
            "model": "Accent",
            "year": "2013",
          },
          "vehicleTitle": "2013 Hyundai Accent",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ListItem
            vehicle={
                        Object {
                                    "bodytype": "sedan",
                                    "created_at": "2016-10-14T20:13:22.586Z",
                                    "drivetrain": "FWD",
                                    "id": 0,
                                    "image_url": "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
                                    "make": "Kia",
                                    "mileage": 24235,
                                    "model": "Optima",
                                    "year": "2013",
                                  }
            }
            vehicleTitle="2013 Kia Optima"
/>,
          <ListItem
            vehicle={
                        Object {
                                    "bodytype": "sedan",
                                    "created_at": "2016-11-14T20:13:22.586Z",
                                    "drivetrain": "FWD",
                                    "id": 1,
                                    "image_url": "http://www.conceptcarz.com/images/Hyundai/2013-Hyundai-Accent-Sedan-Image-01.jpg",
                                    "make": "Hyundai",
                                    "mileage": 21587,
                                    "model": "Accent",
                                    "year": "2013",
                                  }
            }
            vehicleTitle="2013 Hyundai Accent"
/>,
        ],
        "className": "list-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "vehicle": Object {
              "bodytype": "sedan",
              "created_at": "2016-10-14T20:13:22.586Z",
              "drivetrain": "FWD",
              "id": 0,
              "image_url": "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
              "make": "Kia",
              "mileage": 24235,
              "model": "Optima",
              "year": "2013",
            },
            "vehicleTitle": "2013 Kia Optima",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "vehicle": Object {
              "bodytype": "sedan",
              "created_at": "2016-11-14T20:13:22.586Z",
              "drivetrain": "FWD",
              "id": 1,
              "image_url": "http://www.conceptcarz.com/images/Hyundai/2013-Hyundai-Accent-Sedan-Image-01.jpg",
              "make": "Hyundai",
              "mileage": 21587,
              "model": "Accent",
              "year": "2013",
            },
            "vehicleTitle": "2013 Hyundai Accent",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
